Update to webpack 4

mauriciabad opened the pull request:
> This is a fix for issue #23 

Disclaimer: I've never coded a plugin for webpack. I just want to help fix the issue.

I found how a similar problem was solved. So I tried to do the same to this plugin's code. 
https://github.com/jantimon/html-webpack-plugin/issues/875
https://github.com/facebook/create-react-app/pull/4077/files#diff-3b55301108483c3bb5289672dd9caa59

And this is the documentation for the new hooks:
https://webpack.js.org/api/compiler-hooks/

I don't know how to test if the code works, @theGC can you do it? 😄

theGC commented:
@mauriciabad thanks for looking into this. The upgrade is sorely needed, I'm just stuffed for time so I really appreciate any help you can give.

I looked at the Travis build and it's failing on a spread operator within node_modules:

https://travis-ci.org/github/theGC/html-webpack-inline-svg-plugin/jobs/672753352

This seems likely to just be a config issue with the build tests, I'll see if I can find some time this weekend but if you want to look at it further you should be able to runt the tests locally via;

```
npm run test
```

This way you can troubleshoot things prior to pushing a pull request.

mauriciabad commented:
Okay, I updated the Travis node.js version and fixed a thing to let it build successfully.
Take a look whenever you can, no worries.

Now the problem is that it doesn't output anything 😅 .
I found how to use html-webpack-plugin hooks in ther documentation:
https://github.com/jantimon/html-webpack-plugin#beforeemit-hook
But I don't know which hook we need.

AntonioRedondo commented on package.json:
```diff
@@ -31,18 +31,18 @@
     },
     "homepage": "https://github.com/thegc/html-webpack-inline-svg-plugin",
```
On line 3 of `package.json` the version of the package has to be bumped up, perhaps to `1.3.1`?

mauriciabad replied:
Definitely the version should change, but I didn't update it because I didn't know if it would be `1.3.1` or `1.4.0` or even `1.3.2` if another PR is merged before this one.

Should I set it to `1.3.1`?

mauriciabad replied:
I set it to `1.3.1`

theGC commented:
added v2.x.x to target webpack 4

v1.3.0 will remain for webpack 3
